pkg/auth/token_verifier.go

```diff
@@ -51,10 +59,11 @@ type IDToken struct {
 	AccessTokenHash string
 }
 
-func NewOIDCTokenVerifier(ctx context.Context) *OIDCTokenVerifier {
+func NewOIDCTokenVerifier(ctx context.Context, statsReporter filter.StatsReporter) *OIDCTokenVerifier {
 	tokenHandler := &OIDCTokenVerifier{
-		logger:     logging.FromContext(ctx).With("component", "oidc-token-handler"),
-		restConfig: injection.GetConfig(ctx),
+		logger:        logging.FromContext(ctx).With("component", "oidc-token-handler"),
+		restConfig:    injection.GetConfig(ctx),
+		statsReporter: statsReporter,
```

Comment:
Does this looks good? @creydr  By passing the statsReporter to the verifier, and [pass the reportArgs when verifying happens.](https://github.com/knative/eventing/pull/8015/files#diff-5a45b6a786dbd89c11dc49f5d111d8b06700e9770b13a1f37d4896c3625f5a20R164)

Reply:
Can we make this an optional argument (e.g. by using a configuration-option function as we do for example in https://github.com/knative/eventing/blob/825202fdffc5f2b6b71500f232aa0319700aad2b/pkg/reconciler/testing/v1/broker.go#L35-L44 with `type BrokerOption func(*v1.Broker)`)? This allows us to add options, without the need to update the packages immediately which use the `NewOIDCTokenVerifier`.